File: api/src/test/java/net/blueshell/api/factory/model/EventFeedbackFactory.java
package net.blueshell.api.factory.model;

import net.blueshell.api.testutil.ModelTestUtils;
import com.github.javafaker.Faker;
import lombok.RequiredArgsConstructor;
import net.blueshell.api.model.event.EventFeedback;
import org.springframework.stereotype.Component;

import java.util.concurrent.atomic.AtomicLong;

/**
 * Factory for EventFeedback model test instances.
 */
@Component
@RequiredArgsConstructor
public class EventFeedbackFactory {

    private static final AtomicLong counter = new AtomicLong(1000);
    private final Faker faker;
    private final EventFactory eventFactory;

    public EventFeedback createBasic() {
        EventFeedback ef = new EventFeedback();
        ModelTestUtils.setId(ef, generateId());
        ef.setEvent(eventFactory.createBasic());
        ef.setFeedback(faker.lorem().sentence(10));
        return ef;
    }

    public EventFeedback createFull() {
        return createBasic();
    }

    public EventFeedback createWithCustomizations(java.util.function.Consumer<EventFeedback> customizer) {
        EventFeedback ef = createFull();
        customizer.accept(ef);
        return ef;
    }

    private Long generateId() {
        return counter.incrementAndGet();
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/model/event/EventFeedbackFactory.kt
package net.blueshell.api.factory.model.event

import com.github.javafaker.Faker
import net.blueshell.api.event.domain.model.EventFeedback
import org.springframework.stereotype.Component
import java.util.concurrent.atomic.AtomicLong
import java.util.function.Consumer

/**
 * Factory for EventFeedback model test instances.
 */
@Component
class EventFeedbackFactory(
    private val faker: Faker,
    private val eventFactory: EventFactory
) {

    fun createBasic(): EventFeedback {
        val feedback = EventFeedback()
        feedback.event = eventFactory.createBasic()
        feedback.feedback = faker.lorem().sentence(10)
        return feedback
    }

    fun createFull(): EventFeedback = createBasic()

    fun createWithCustomizations(customizer: Consumer<EventFeedback>): EventFeedback {
        val feedback = createFull()
        customizer.accept(feedback)
        return feedback
    }

    private fun generateId(): Long = COUNTER.incrementAndGet()

    private companion object {
        val COUNTER = AtomicLong(1000)
    }
}
